### PR TITLE
Rework left-hand nav order

### LINKED ISSUE #338: Reorder the left-hand side navigation structure
### Feature request

The current structure does not follow a logical order. Restructure it to make it more user friendly. More details can be found in the roadmap updates in #337, which in turn is based on our discussions in the Docs WG.

### PR DESCRIPTION
**Pull request by geriom**
# Changes
A less intrusive attempt at reordering the left nav tree. This way we don't have to make any changes on each component repo, nor fundamentally change the way we build the site.

- Refactor config file for legibility
- Write new menu logic to pick manual menu from config file
- Replace spaces with dashes in folder names

/closes #338 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/website/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Review by dibyom — APPROVED** [review]
Ordering makes sense to me! 

(Don't know a lot about the other changes in config.yaml so will wait for someone else to lgtm)

File: config.yaml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    - name: "Results"
+      url: "/docs/results/"
+      weight: 11 
+    - name: "Contribute"
+      url: "/docs/contribute/"
+      weight: 100 
+
+        
+
+
```
Comment (afrittoli) [inline config.yaml:111]: NIT: empty lines

File: config.yaml [hunk 2]
```diff
@@ -0,0 +1,111 @@
+baseURL: /
```
[the remark this answers is not in the document]
Comment (geriom) [reply config.yaml:1]: There's no functional difference. In my opinion it's more human-readable than toml and the syntax enforces keeping options for the same stuff together.

Docsy migrated from tom to yaml a few months ago, doing the same might make is easier to spot new/deprecated options.

That being said, there's no technical reason to migrate. Happy to revert to TOML if others disagree.

File: config.yaml (hunk 1 above)
Comment (geriom) [reply config.yaml:111]: Deleted empty lines.  I guess while I was adding new sections kept inserting new lines every time :(

**Review by afrittoli — COMMENTED** [review]
/lgtm

**Pull request merged**